docs/workloads/azure/data/getting_started/etl_pipelines_deployment_azure.md

```diff
@@ -1,17 +1,123 @@
+  - amido-stacks-de-pipeline-prod
+  - amido-stacks-infra-credentials-prod
+  - stacks-credentials-prod-kv
+
+# Datalake containers
+
+bronze_container: raw
+silver_container: staging
+gold_container: curated
+```
+
+## Generate project artifacts using Datastacks
+
+Use the Datastacks CLI to generate the artifacts for the new workload, using the prepared config file (replacing `path_to_config_file/my_config.yaml` with the appropriate path). Note, a workload with Data Quality steps requires a data platform with a Databricks workspace:
+
+```bash
+# Initiate Datastacks using poetry:
+poetry run datastacks
+
+# Generate resources for an ingest pipeline (without Data Quality steps)
+datastacks generate ingest --config="path_to_config_file/my_config.yaml"
+
+# Generate resources for an ingest pipeline (with added Data Quality steps)
+datastacks generate ingest --config="path_to_config_file/my_config.yaml" --data-quality
+```
+
+This should add new project artifacts for the workload under `de_workloads/ingest`, based on the ingest workload templates. Review the resources that have been generated.
+
+Once reviewed, commit the resources and push the branch to the remote project repository.
+
+## Deploy new workload in non-prod environment
+
+The generated workload will contain a YAML file containing a template Azure DevOps CI/CD pipeline for the workload, named `de-ingest-ado-pipeline.yaml`. This can be added as the definition for a new pipeline in Azure DevOps.
+1. Sign-in to your Azure DevOps organization and go to your project.
+2. Go to Pipelines, and then select New pipeline.
+3. Name the new pipeline to match the name of your new workload, e.g. `de-ingest-azuresql-mynewexample`
+4. For the pipeline definition, specify the YAML file in the project repository feature branch (e.g. `de-ingest-ado-pipeline.yaml`)
+5. Save and run the pipeline.
+
+Running this pipeline in Azure DevOps will deploy the artifacts into the non-prod environment and run tests. If successful, the generated resources will now be available in the non-prod Stacks environment.
+
+Continue to make any further amendments required to the new workload, re-running the pipeline as required - for example:
+* Update the (`ingest_config`) file with details of the data required from the new data source
+* (If including DQ) update the (`ingest_dq`) file with details of data quality checks required on the data
+* Update end-to-end tests, to ensure test coverage of the new data pipeline
 
-## TODO
+## Deploy new workload in further environments
 
-## ETL pipelines deployment
+It is recommended in any Stacks data platform that processes for deploying and releasing to further should be agreed and documented, ensuring sufficient review and quality assurance of any new workloads. The template CI/CD pipelines provided are based upon two platform environments (non-prod and prod) - but these may be amended depending upon the specific requirements of your project and organisation.
```

Comment:
These prod and nonprod environments are very much enforced in Stacks... Changing the names of the environments would require lots of manual updates and changing the number of environments even more significant changes. Maybe at some point we should tackle that. Feel free to resolve.

https://stacks.amido.com/docs/infrastructure/azure/pipelines/azure_devops

Reply:
I suggest we add a story for this.